tee/tf/main.tf

```diff
@@ -0,0 +1,277 @@
+  name = "ec2-nitro-role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action    = "sts:AssumeRole",
+      Effect    = "Allow",
+      Principal = { Service = "ec2.amazonaws.com" }
+    }]
+  })
+}
+
+resource "aws_iam_instance_profile" "ec2_profile" {
+  name = "ec2-nitro-profile"
+  role = aws_iam_role.ec2_role.name
+}
+
+
+# EC2 Instance for Nitro Enclave + Proxy
+resource "aws_instance" "proxy_instance" {
+  ami                    = data.aws_ami.al2023.id
+  instance_type          = var.instance_type
+  key_name               = length(var.key_name) > 0 ? var.key_name : null
+  vpc_security_group_ids = [aws_security_group.proxy_sg.id]
+  subnet_id              = data.aws_subnets.public.ids[0]
+  
+  iam_instance_profile        = aws_iam_instance_profile.ec2_profile.name
+  associate_public_ip_address = true
+  
+  enclave_options {
+    enabled = true
+  }
+
+  user_data = base64encode(<<-EOF
+#!/bin/bash
+set -eux
+
+# Log everything to a file for debugging
+exec > >(tee /var/log/user-data.log) 2>&1
+
+echo "Starting Shielder Prover Server setup..."
+
+# Update system and install required packages
+dnf update -y
+dnf install -y aws-nitro-enclaves-cli aws-nitro-enclaves-cli-devel docker jq --allowerasing
+
+# Add ec2-user to required groups
+usermod -aG ne ec2-user
+usermod -aG docker ec2-user
+
+# Configure Nitro Enclaves allocator
+cat > /etc/nitro_enclaves/allocator.yaml << 'EOL'
+---
+# Enclave configuration file
+#
+# How much memory to allocate for enclaves (in MiB).
+memory_mib: 2048
+#
+# How many CPUs to reserve for enclaves.
+cpu_count: 2
+EOL
+
+# Enable and start services
+systemctl enable nitro-enclaves-allocator.service
+systemctl start nitro-enclaves-allocator.service
+systemctl enable docker
+systemctl start docker
+
+# Wait for services to be ready
+sleep 10
+
+# Download the enclave image
+echo "Downloading enclave image from ${var.eif_url}"
+curl -sSL "${var.eif_url}" -o /home/ec2-user/image.eif
+chown ec2-user:ec2-user /home/ec2-user/image.eif
+
+# Set TEE_CID
+export TEE_CID=16
+
+# Run the enclave
+echo "Starting Nitro Enclave with CID=$${TEE_CID}"
+nitro-cli run-enclave --cpu-count 2 --memory 2048 --enclave-cid $${TEE_CID} --eif-path /home/ec2-user/image.eif${var.debug_mode ? " --debug-mode" : ""}
+
+# Wait for enclave to start
+sleep 5
+
+# Verify enclave is running
+nitro-cli describe-enclaves
+
+# Get the actual enclave CID (should be 16, but let's be safe)
+ACTUAL_CID=$(nitro-cli describe-enclaves | jq -r '.[0].EnclaveCID // empty')
+if [ -z "$ACTUAL_CID" ]; then
+  echo "ERROR: Failed to get enclave CID"
+  exit 1
+fi
+
+echo "Enclave running with CID=$${ACTUAL_CID}"
+
+# Start the proxy Docker container
+echo "Starting proxy container with image ${var.proxy_image}"
+docker run -d \
+  --name shielder-prover-proxy \
+  --restart always \
+  -p 3000:3000 \
+  --security-opt seccomp=unconfined \
+  -e ENCLAVE_CID="$${ACTUAL_CID}" \
+  -e RUST_LOG=debug \
```

Comment:
Can we please parametrize this as well? Sometimes we'd need TRACE here (as `axum` handlers are instrumented).

Reply:
I'll create a task for it